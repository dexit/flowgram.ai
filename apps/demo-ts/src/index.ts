export * from './types'
export * from './runtime'
export * from './translate'
export * from './validate'
export * from './util'
