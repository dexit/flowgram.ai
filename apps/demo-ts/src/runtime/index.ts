export * from './run'
export * from './RuntimeError'
