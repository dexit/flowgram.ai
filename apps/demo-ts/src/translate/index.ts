export * from './nodes'
export * from './translate'
