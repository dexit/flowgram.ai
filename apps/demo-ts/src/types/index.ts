export * from '@flow-ide-editor/schema'
export * from './nodes'

export class InvalidSchemaError extends Error {}
export class InvalidNodeTypeError extends Error {}
export class InvalidTypeError extends Error {}
export class NotImplementedError extends Error {}

export type FlowArgs = Record<string, unknown>
