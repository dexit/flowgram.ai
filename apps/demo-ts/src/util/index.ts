export * from './log'
export * from './JSBuffer'
