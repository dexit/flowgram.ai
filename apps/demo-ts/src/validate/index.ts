import { FlowNodeSchema } from '@flow-ide-editor/schema'
import { InvalidSchemaError } from '../types'

// TODO use json-schema validation
export function validateOrThrow(node: FlowNodeSchema): void {
  if (!node.definition) throw new InvalidSchemaError()
  if (!node.definition.info) throw new InvalidSchemaError()
  if (!node.definition.info.title) throw new InvalidSchemaError()
}
