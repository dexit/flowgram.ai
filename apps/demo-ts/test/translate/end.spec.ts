import { EndNodeSchema, translateNode } from '../../src'

describe('translate', () => {
  it('End - 0 param', () => {
    const node: EndNodeSchema = {
      id: 'end',
      type: 'end',
      definition: {
        info: { title: 'end', description: '', icon: '' },
        inputs: {
          type: 'object',
          properties: {},
        },
        script: { language: 'python', content: '' },
      },
      data: {},
    }

    expect(translateNode(node)).toMatchSnapshot()
  })

  it('End - 1 param', () => {
    const node: EndNodeSchema = {
      id: 'end',
      type: 'end',
      definition: {
        info: { title: 'end', description: '', icon: '' },
        inputs: {
          type: 'object',
          properties: {
            data: { type: 'string' },
          },
        },
        script: { language: 'python', content: '' },
      },
      data: {
        data: { type: 'expression', content: 'start.outputs.query' },
      },
    }

    expect(translateNode(node)).toMatchSnapshot()
  })

  it('End - mores params', () => {
    const node: EndNodeSchema = {
      id: 'end',
      type: 'end',
      definition: {
        info: { title: 'end', description: '', icon: '' },
        inputs: {
          type: 'object',
          properties: {
            v_string: { type: 'string' },
            v_integer: { type: 'integer' },
            v_number: { type: 'number' },
            v_boolean: { type: 'boolean' },
            v_array: { type: 'array', items: { type: 'string' } },
            v_object: { type: 'object' },
            v_map: { type: 'map', additionalProperties: { type: 'string' } },
          },
        },
        script: { language: 'python', content: '' },
      },
      data: {
        v_string: { type: 'expression', content: "''" },
        v_integer: { type: 'expression', content: '10' },
        v_number: { type: 'expression', content: '3.14' },
        v_boolean: { type: 'expression', content: 'True' },
        v_array: { type: 'expression', content: "['a','b']" },
        v_object: { type: 'expression', content: "{'a':'abc'}" },
        v_map: { type: 'expression', content: "{'b':'xyz'}" },
      },
    }

    expect(translateNode(node)).toMatchSnapshot()
  })
})
