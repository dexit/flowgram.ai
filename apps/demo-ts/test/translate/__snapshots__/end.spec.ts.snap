// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`translate End - 0 param 1`] = `
"  //=== end@end | {level:1,blockIndex:0}
  const end = {}
  end.data = {
  }
  return end.data
  "
`;

exports[`translate End - 1 param 1`] = `
"  //=== end@end | {level:1,blockIndex:0}
  const end = {}
  end.data = {
    data: start.outputs.query,
  }
  return end.data
  "
`;

exports[`translate End - mores params 1`] = `
"  //=== end@end | {level:1,blockIndex:0}
  const end = {}
  end.data = {
    v_string: '',
    v_integer: 10,
    v_number: 3.14,
    v_boolean: True,
    v_array: ['a','b'],
    v_object: {'a':'abc'},
    v_map: {'b':'xyz'},
  }
  return end.data
  "
`;
